zebra-consensus/src/transaction/tests.rs

```diff
@@ -1542,3 +1544,173 @@ fn add_to_sprout_pool_after_nu() {
         Ok(())
     );
 }
+
+#[test]
+fn coinbase_outputs_are_decryptable_for_historical_blocks() -> Result<(), Report> {
+    zebra_test::init();
+
+    coinbase_outputs_are_decryptable_for_historical_blocks_for_network(Network::Mainnet)?;
+    coinbase_outputs_are_decryptable_for_historical_blocks_for_network(Network::Testnet)?;
+
+    Ok(())
+}
+
+fn coinbase_outputs_are_decryptable_for_historical_blocks_for_network(
+    network: Network,
+) -> Result<(), Report> {
+    let block_iter = match network {
+        Network::Mainnet => zebra_test::vectors::MAINNET_BLOCKS.iter(),
+        Network::Testnet => zebra_test::vectors::TESTNET_BLOCKS.iter(),
+    };
+
+    let mut tested_coinbase_txs = 0;
+    let mut tested_non_coinbase_txs = 0;
+
+    for (_height, block) in block_iter {
+        let block = block
+            .zcash_deserialize_into::<Block>()
+            .expect("block is structurally valid");
+        let height = block.coinbase_height().expect("a valid height");
```

Comment:
can we use the unused height from the loop here instead?

Reply:
It's just that the height from the loop is a `u32` and here it needs a `Height`. But indeed it's less confusing to just convert it instead. Done in 52c128a42